compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -140,7 +150,7 @@ pub fn intern_const_alloc_recursive<
     ecx: &mut InterpCx<'mir, 'tcx, M>,
     intern_kind: InternKind,
     ret: &MPlaceTy<'tcx>,
-) -> Result<(), ErrorGuaranteed> {
+) -> Result<(), InternResult> {
```

Comment:
This is a strange return type, what would `Err(InternResult::default())` mean?

I would have expected `InternResult` to be an `enum`, not a `struct`.